providers/supermicro/supermicro.go

```diff
@@ -300,6 +320,11 @@ func (c *Client) bmcQueryor(ctx context.Context) (bmcQueryor, error) {
 			return nil, errors.Wrap(ErrModelUnknown, err.Error())
 		}
 
+		// ensure the device model matches the expected queryor
+		if !expected(deviceModel, bmc) {
```

Comment:
include a `bmc.Close()` before the continue here to ensure we don't leave a successful connection open